src/link/edit.js

```diff
@@ -102,6 +97,7 @@ const Edit = ( {
 					onRemove={ removeAttributes }
 					value={ value }
 					activeAttributes={ activeAttributes }
+					onClose={ stopAddingPreview }
```

Comment:
@stephanebisson when you are testing this, does it close the popup in mobile UI?